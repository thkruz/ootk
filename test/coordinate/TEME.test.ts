import { ClassicalElements, EpochUTC, J2000, Kilometers, KilometersPerSecond, Radians, TEME, Vector3D }
  from './../../src/main';
describe('TEME', () => {
  let stateVector: TEME;

  beforeEach(() => {
    stateVector = new J2000(
      EpochUTC.fromDateTime(new Date(1705109326817)),
      new Vector3D(1538.223335842895 as Kilometers, 5102.261204021967 as Kilometers, 4432.634965003577 as Kilometers),
      new Vector3D(
        -7.341518909379302 as KilometersPerSecond,
        0.6516718453998644 as KilometersPerSecond,
        1.7933882499861993 as KilometersPerSecond,
      ),
    ).toTEME();
  });

  // name
  it('should return the name of the coordinate system', () => {
    expect(stateVector.name).toMatchSnapshot();
  });

  // inertial
  it('should return whether the coordinate system is inertial', () => {
    expect(stateVector.inertial).toMatchSnapshot();
  });

  // fromClassicalElements
  it('should create a TEME coordinate from classical orbital elements', () => {
    const exampleDate = new Date(1705109326817);
    const epoch = EpochUTC.fromDateTime(exampleDate);
    const elements = new ClassicalElements({
      epoch,
      semimajorAxis: 6943.547853722985 as Kilometers,
      eccentricity: 0.0011235968124658146,
      inclination: 0.7509087232045765 as Radians,
      rightAscension: 0.028239555738616327 as Radians,
      argPerigee: 2.5386411901807353 as Radians,
      trueAnomaly: 0.5931399364974058 as Radians,
    });

    expect(TEME.fromClassicalElements(elements)).toMatchSnapshot();
  });

  // toJ2000
  it('should convert the TEME coordinates to J2000 coordinates', () => {
    expect(stateVector.toJ2000()).toMatchSnapshot();
  });
});
