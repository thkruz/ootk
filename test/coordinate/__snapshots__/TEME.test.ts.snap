// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEME should convert the TEME coordinates to J2000 coordinates 1`] = `
J2000 {
  "epoch": EpochUTC {
    "posix": 1705109326.817,
  },
  "position": Vector3D {
    "x": 1538.2233358428946,
    "y": 5102.2612040219665,
    "z": 4432.634965003576,
  },
  "velocity": Vector3D {
    "x": -7.341518909379303,
    "y": 0.6516718453998649,
    "z": 1.793388249986199,
  },
}
`;

exports[`TEME should create a TEME coordinate from classical orbital elements 1`] = `
TEME {
  "epoch": EpochUTC {
    "posix": 1705109326.817,
  },
  "position": Vector3D {
    "x": -6935.3813042612455,
    "y": -146.1261328371035,
    "z": 46.43907991639276,
  },
  "velocity": Vector3D {
    "x": 0.07740377357485445,
    "y": -5.543955093986134,
    "z": -5.174123893746047,
  },
}
`;

exports[`TEME should return the name of the coordinate system 1`] = `"TEME"`;

exports[`TEME should return whether the coordinate system is inertial 1`] = `true`;
